### PR TITLE
feat(store-engine-sqleet): Remove database directory on purge

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: packages/store-engine-sqleet/src/SQLeetEngine.ts [hunk 1]
```diff
@@ -106,6 +106,14 @@ export class SQLeetEngine implements CRUDEngine {
     if (this.db) {
       await this.db.close();
       await this.db.wipe(this.storeName);
+      if (this.nodeDatabaseDir) {
+        fs.rmdirSync(this.nodeDatabaseDir, {recursive: true});
```
Comment (ffflorian) [inline packages/store-engine-sqleet/src/SQLeetEngine.ts]: Why not use `async` here, too?

```suggestion
        await fs.rmdir(this.nodeDatabaseDir, {recursive: true});
```

Comment (AndyLnd) [reply packages/store-engine-sqleet/src/SQLeetEngine.ts]: This is not using `fs-extra` so rmdir doesn't return a promise.

Comment (ffflorian) [reply packages/store-engine-sqleet/src/SQLeetEngine.ts]: Ah sorry, wrong suggested change. Here is the correct one:
```suggestion
        await fs.promises.rmdir(this.nodeDatabaseDir, {recursive: true});
```

Comment (bennyn) [reply packages/store-engine-sqleet/src/SQLeetEngine.ts]: Why is ` await fs.promises.rmdir` better than `fs.rmdirSync`?

Comment (ffflorian) [reply packages/store-engine-sqleet/src/SQLeetEngine.ts]: Because `fs.rmdirSync` will block the whole I/O while the `async` solution is non-blocking.

**Pull request closed without merging**